src/fidesops/models/email.py

```diff
@@ -0,0 +1,101 @@
+from fidesops.schemas.email.email import EmailServiceType, EmailServiceSecretsMailgun, SUPPORTED_EMAIL_SERVICE_SECRETS, \
+    EmailPurpose
+from fidesops.schemas.email.email_secrets_docs_only import possible_email_secrets
+
+logger = logging.getLogger(__name__)
+
+
+def get_schema_for_secrets(
+        service_type: EmailServiceType,
+        secrets: possible_email_secrets,
+) -> SUPPORTED_EMAIL_SERVICE_SECRETS:
+    """
+    Returns the secrets that pertain to `service_type` represented as a Pydantic schema
+    for validation purposes.
+    """
+    try:
+        schema = {
+            EmailServiceType.MAILGUN: EmailServiceSecretsMailgun,
+        }[service_type]
+    except KeyError:
+        raise ValueError(
+            f"`service_type` {service_type} has no supported `secrets` validation."
+        )
+
+    try:
+        return schema.parse_obj(secrets)  # type: ignore
+    except ValidationError as exc:
+        # Pydantic requires validators raise either a ValueError, TypeError, or AssertionError
+        # so this exception is cast into a `ValueError`.
+        errors = [f"{err['msg']} {str(err['loc'])}" for err in exc.errors()]
+        raise ValueError(errors)
+
+
+class EmailConfig(Base):
+    """The DB ORM model for EmailConfig"""
+    key = Column(String, index=True, unique=True, nullable=False)
+    name = Column(String, unique=True, index=True)
+    service_type = Column(Enum(EmailServiceType), index=True, nullable=False)
+    details = Column(MutableDict.as_mutable(JSONB), nullable=False)
+    secrets = Column(
+        MutableDict.as_mutable(
+            StringEncryptedType(
+                JSONTypeOverride,
+                config.security.app_encryption_key,
+                AesGcmEngine,
+                "pkcs5",
+            )
+        ),
+        nullable=True,
+    )  # Type bytea in the db
+
+    def set_secrets(
+            self,
+            *,
+            db: Session,
+            email_secrets: possible_email_secrets,
+    ) -> None:
+        """Creates or updates secrets associated with a config id"""
+
+        service_type = self.service_type
+        if not service_type:
+            raise ValueError("This object must have a `service_type` to validate secrets.")
+
+        try:
+            get_schema_for_secrets(
+                service_type=service_type,  # type: ignore
+                secrets=email_secrets,
+            )
+        except (
+                KeyError,
+                ValidationError,
+        ) as exc:
+            logger.error("Error: %s", exc)
+            # We don't want to handle these explicitly here, only in the API view
+            raise
+
+        self.secrets = email_secrets
+        self.save(db=db)
+
+
+class EmailConfigPurpose(Base):
+    """The DB ORM model for Email purpose"""
+
+    config_key = Column(String, ForeignKey(EmailConfig.key), index=True, unique=False, nullable=False)
+    purpose = Column(Enum(EmailPurpose), index=True, unique=True, nullable=False)
```

Comment:
If we'll only support unique purposes across the system, why do we need an extra table here?

Reply:
yup, since we discussed approach, I'm removing the concept of email purpose entirely